pkg/controller/direct/sql/mapping.go

```diff
@@ -793,10 +797,43 @@ func Convert_SQLInstance_API_v1_To_KRM_status(in *api.DatabaseInstance, out *krm
 	return nil
 }
 
-func LazyPtr[T comparable](v T) *T {
-	var defaultValue T
-	if v == defaultValue {
-		return nil
+func SQLInstanceKRMToGCPCloneRequest(in *krm.SQLInstance, refs *SQLInstanceInternalRefs) (*api.InstancesCloneRequest, error) {
+	if in == nil {
+		return nil, fmt.Errorf("cannot convert nil SQLInstance")
+	}
+
+	if in.Spec.CloneSource == nil {
+		return nil, fmt.Errorf("cannot convert nil CloneSource")
```

Comment:
Similarly, I wondered if the error message can be more verbose and actionable. E.g. `cannot convert nil CloneSource, CloudSource is required`.

Reply:
Same here, this error would only happen if developer misuses the function. I'll add a comment.

Reply:
fixed